cpp/src/Ice/CMakeLists.txt

```diff
@@ -0,0 +1,595 @@
+	${GENERATED_SLICES_DIR}/Ice/LocatorF.h
+	${GENERATED_SLICES_DIR}/Ice/Logger.h
+	${GENERATED_SLICES_DIR}/Ice/LoggerF.h
+	${GENERATED_SLICES_DIR}/Ice/Metrics.h
+	${GENERATED_SLICES_DIR}/Ice/ObjectAdapter.h
+	${GENERATED_SLICES_DIR}/Ice/ObjectAdapterF.h
+	${GENERATED_SLICES_DIR}/Ice/ObjectFactory.h
+	${GENERATED_SLICES_DIR}/Ice/Plugin.h
+	${GENERATED_SLICES_DIR}/Ice/PluginF.h
+	${GENERATED_SLICES_DIR}/Ice/Process.h
+	${GENERATED_SLICES_DIR}/Ice/ProcessF.h
+	${GENERATED_SLICES_DIR}/Ice/Properties.h
+	${GENERATED_SLICES_DIR}/Ice/PropertiesAdmin.h
+	${GENERATED_SLICES_DIR}/Ice/PropertiesF.h
+	${GENERATED_SLICES_DIR}/Ice/RemoteLogger.h
+	${GENERATED_SLICES_DIR}/Ice/Router.h
+	${GENERATED_SLICES_DIR}/Ice/RouterF.h
+	${GENERATED_SLICES_DIR}/Ice/ServantLocator.h
+	${GENERATED_SLICES_DIR}/Ice/ServantLocatorF.h
+	${GENERATED_SLICES_DIR}/Ice/SliceChecksumDict.h
+	${GENERATED_SLICES_DIR}/Ice/ValueFactory.h
+	${GENERATED_SLICES_DIR}/Ice/Version.h
+)
+
+set(GENERATED_ICE_SOURCES
+	${GENERATED_SLICES_DIR}/Ice/BuiltinSequences.cpp
+	${GENERATED_SLICES_DIR}/Ice/Communicator.cpp
+	${GENERATED_SLICES_DIR}/Ice/CommunicatorF.cpp
+	${GENERATED_SLICES_DIR}/Ice/Connection.cpp
+	${GENERATED_SLICES_DIR}/Ice/ConnectionF.cpp
+	${GENERATED_SLICES_DIR}/Ice/Current.cpp
+	${GENERATED_SLICES_DIR}/Ice/Endpoint.cpp
+	${GENERATED_SLICES_DIR}/Ice/EndpointF.cpp
+	${GENERATED_SLICES_DIR}/Ice/EndpointTypes.cpp
+	${GENERATED_SLICES_DIR}/Ice/FacetMap.cpp
+	${GENERATED_SLICES_DIR}/Ice/Identity.cpp
+	${GENERATED_SLICES_DIR}/Ice/ImplicitContext.cpp
+	${GENERATED_SLICES_DIR}/Ice/ImplicitContextF.cpp
+	${GENERATED_SLICES_DIR}/Ice/Instrumentation.cpp
+	${GENERATED_SLICES_DIR}/Ice/InstrumentationF.cpp
+	${GENERATED_SLICES_DIR}/Ice/LocalException.cpp
+	${GENERATED_SLICES_DIR}/Ice/Locator.cpp
+	${GENERATED_SLICES_DIR}/Ice/LocatorF.cpp
+	${GENERATED_SLICES_DIR}/Ice/Logger.cpp
+	${GENERATED_SLICES_DIR}/Ice/LoggerF.cpp
+	${GENERATED_SLICES_DIR}/Ice/Metrics.cpp
+	${GENERATED_SLICES_DIR}/Ice/ObjectAdapter.cpp
+	${GENERATED_SLICES_DIR}/Ice/ObjectAdapterF.cpp
+	${GENERATED_SLICES_DIR}/Ice/ObjectFactory.cpp
+	${GENERATED_SLICES_DIR}/Ice/Plugin.cpp
+	${GENERATED_SLICES_DIR}/Ice/PluginF.cpp
+	${GENERATED_SLICES_DIR}/Ice/Process.cpp
+	${GENERATED_SLICES_DIR}/Ice/ProcessF.cpp
+	${GENERATED_SLICES_DIR}/Ice/Properties.cpp
+	${GENERATED_SLICES_DIR}/Ice/PropertiesAdmin.cpp
+	${GENERATED_SLICES_DIR}/Ice/PropertiesF.cpp
+	${GENERATED_SLICES_DIR}/Ice/RemoteLogger.cpp
+	${GENERATED_SLICES_DIR}/Ice/Router.cpp
+	${GENERATED_SLICES_DIR}/Ice/RouterF.cpp
+	${GENERATED_SLICES_DIR}/Ice/ServantLocator.cpp
+	${GENERATED_SLICES_DIR}/Ice/ServantLocatorF.cpp
+	${GENERATED_SLICES_DIR}/Ice/SliceChecksumDict.cpp
+	${GENERATED_SLICES_DIR}/Ice/ValueFactory.cpp
+	${GENERATED_SLICES_DIR}/Ice/Version.cpp
+)
+
+if(MSVC AND NOT BUILD_ICE_UWP)
+	set(GENERATED_ICE_RESOURCES
+		${GENERATED_SLICES_DIR}/Ice/EventLoggerMsg.h
+		${GENERATED_SLICES_DIR}/Ice/EventLoggerMsg.rc
+	)
+endif()
+
+if(BUILD_ICE_UWP)
```

Comment:
Unless you need UWP (quite unlikely), I suggest to simplify and not include UWP support in this cmake build system.

Reply:
I didn't need it. I just wanted to stay consistent with the solutions and projects in the source. I believe it works, also.

Reply:
I guess you meant vcpkg recognizes uwp as a target platform.

I would just drop it: Ice 3.7.3 did not include UWP binaries, and this UWP support in the cmake build system is not free - it adds various file lists, conditionals, variables.

Reply:
Yes, vcpkg has uwp as a target option. It doesn't have to have uwp and I can add a setting to vcpkg that turns off uwp builds for Ice.

Reply:
This has been addressed.